### PR TITLE
[CRAW-363] Configuring adjusted score for URLs from different sources.

### PR DESCRIPTION
**Pull request by vanchpuck**

### CODE REVIEW

File: src/plugin/prioritising-input/src/java/com/integralads/scoring/InputFileScoringFilter.java [hunk 1]
```diff
@@ -23,17 +26,17 @@
...
 public class InputFileScoringFilter implements ScoringFilter {
 
-  public static final String INPUT_FILE_MASK = "input.file.mask";
-  private static final String DEFAULT_FILE_MASK = "";
+  public static final String INPUT_FILE_SCORE = "input.file.score";
+  private static final String DEFAULT_FILE_SCORE = "";
   
   private Configuration conf;
-  private Pattern pattern;
+  private Map<String, Float> scoreMap;
```
Comment (kgusakov) [inline src/plugin/prioritising-input/src/java/com/integralads/scoring/InputFileScoringFilter.java]: Maybe will be better to use `Map<Pattern, Float> scoreMap` and precompile all patterns in `setConf` as you made earlier with one Pattern

File: src/plugin/prioritising-input/src/java/com/integralads/scoring/InputFileScoringFilter.java [hunk 2]
```diff
@@ -83,5 +91,22 @@ public float indexerScore(String url, NutchDocument doc, WebPage page,
       float initScore) throws ScoringFilterException {
     return initScore;
   }
+  
+  private Map<String, Float> readScore(String scoreString) {
+    if(scoreString == null) {
+      throw new IllegalArgumentException("Score string can't be null.");
+    }
+    Map<String, Float> scoreMap = new HashMap<>();
+    String[] scoring = scoreString.split("/");
```
Comment (kgusakov) [inline src/plugin/prioritising-input/src/java/com/integralads/scoring/InputFileScoringFilter.java]: I think that format `(pattern):score|(another-pattern):another-score' will be more human readable

File: src/plugin/prioritising-input/src/java/com/integralads/scoring/InputFileScoringFilter.java [hunk 3]
```diff
@@ -50,9 +53,14 @@ public Configuration getConf() {
...
     Utf8 mark = Mark.INPUT_MARK.checkMark(page);
-    if(mark != null && pattern.matcher(mark.toString()).find() ) {
-      page.setScore(Float.MAX_VALUE);
-      return;
+    if(mark != null) {
+      for(String regexp : scoreMap.keySet()) {
+        Pattern pattern = Pattern.compile(regexp);
+        Matcher matcher = pattern.matcher(mark.toString());
+        if(matcher.find()) {
+          page.setScore(scoreMap.get(regexp));
```
Comment (SergeyWeiss) [inline src/plugin/prioritising-input/src/java/com/integralads/scoring/InputFileScoringFilter.java]: If mark matches several patterns, then page score would be score of last pattern. Shouldn't it be max or sum of scores?

Comment (vanchpuck) [inline src/plugin/prioritising-input/src/java/com/integralads/scoring/InputFileScoringFilter.java]: I think the score of last pattern would be Ok in that case.

Comment (SergeyWeiss) [inline src/plugin/prioritising-input/src/java/com/integralads/scoring/InputFileScoringFilter.java]: I'd ask Alexey if I were you, just in case

**Comment by SergeyWeiss** [comment]
:+1:

File: src/plugin/prioritising-input/src/test/com/integralads/scoring/TestInputFileScoringFilter.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  private static final double DELTA = 1e-15;
+  private static final String URL = "test.info";
+  
+  private InputFileScoringFilter filter = new InputFileScoringFilter();
+  private WebPage page;
+  
+  @Before
+  public void setUp() throws Exception {
+    Configuration conf = new Configuration();
+    conf.set(InputFileScoringFilter.INPUT_FILE_SCORE, "[{'mask':'some','score':'1000'},{'mask':'another','score':'100'}]");
```
Comment (kgusakov) [inline src/plugin/prioritising-input/src/test/com/integralads/scoring/TestInputFileScoringFilter.java]: Let's inject some real masks here, for cheap checking that symbols from regex (like ^/_) are not failing json parser.

**Pull request merged**